### DISCUSSION TITLE
DBIO.sequence and SQL statements batching

### DISCUSSION THREAD

**Discussion by sebarys** [slick/slick#3070] [Questions]
Is it possible to specify for `DBIO.sequence` to use SQL batch https://www.baeldung.com/jdbc-batch-processing or maybe it is done out of the box when all `DBIO`s are interactions with DB (queries) ?

I would like make sure that using `DBIO.sequence` with big number of elements won't cause multiple interactions with DB, but instead everything will be provided to DB engine together in single batch

**Comment by nafg** [comment]
Can those APIs be used with `SELECT` or other queries that return something other than an update count?

If not, then `DBIO.sequence`'s signature cannot be used. Perhaps `DBIO.seq` could be made to support it, but if queries are not allowed then it would need some way to know whether batch mode is feasible. Perhaps inserts and updates are their own subtypes of DBIO so it could use that information. Otherwise perhaps a new API is needed.

Another question is whether this behavior should be used unconditionally, opt-out, or opt-in.

In any case you are invited to send a PR ;)
